### PR TITLE
deps: yeoman-generator@0.24.1 and use promises

### PR DESCRIPTION
**Pull request by tunniclm**

### CODE REVIEW

File: refresh/index.js [hunk 1]
```diff
@@ -673,74 +671,59 @@ module.exports = generators.Base.extend({
...
+              this.env.error(chalk.red('Could not extract title from Swagger API.'));
+            }
+
+            var serverSDKName = response.loaded.info.title.replace(/ /g, '_') + '_ServerSDK';
+            return performSDKGenerationAsync(serverSDKName, 'server_swift', JSON.stringify(response.loaded))
+              .then(generatedID => getServerSDKAsync(serverSDKName, generatedID))
+              .then(sdk => {
+                this.sdkTargets.push(serverSDKName);
+                if (sdk.packages.length > 0) {
+                  this.sdkPackages = sdk.packages; // FIXME! Seems bad
```
Comment (tunniclm) [inline refresh/index.js]: Seems like this only takes the packages from the latest SDK that was generated (provided it had any dependencies at all). Do all server SDKs always have the same dependencies? Or should we really be merging and de-duplicating them?

Comment (tfrank64) [reply refresh/index.js]: Yes, all swift server SDKs have the same dependencies at this point so I did not implement any merging. We could just as well take the first SDK's dependencies and leave it at that.

**Review by tfrank64 — APPROVED** [review]
Took some time to determine where everything moved around, but the changes make sense and look better with Promises.

**Comment by tunniclm** [comment]
Rebased and separated changes into 3 commits with proper descriptions.

**Comment by tunniclm** [comment]
Changed the `FIXME` comment into an explanation comment.

**Review by nhardman — APPROVED** [review]

**Pull request merged**